app/models/innovationTier.server.ts

```diff
@@ -0,0 +1,58 @@
+import { prisma } from "~/db.server"
+
+export type { InnovationTiers } from "@prisma/client"
+
+type newInnovationTier = {
+  name: string
+  benefits: string
+  requisites: string
+  goals: string
+}
+
+export async function getInnovationTiers() {
+  const InnovationTiers = await prisma.innovationTiers.findMany({
+    orderBy: {
+      name: "asc",
+    },
+  })
+  return InnovationTiers
+}
+
+export async function addInnovationTier(input: newInnovationTier) {
+  try {
+    const innovationTier = await prisma.innovationTiers.create({ data: input })
+    return { innovationTier, error: "" }
+  } catch (e) {
+    throw new Error(JSON.stringify(e))
+  }
+}
+
+export async function removeInnovationTier({ name }: { name: string }) {
+  try {
+    const innovationTier = await prisma.innovationTiers.findFirst({
+      where: { name },
+    })
+    if (!innovationTier) {
+      return { error: "Project Status not found in DB" }
```

Comment:
Hi @LuisTejedaS!

Thank you for the feedback! I have a couple of comments about it that maybe we can discuss in the standup.
If I throw an error then the Component crashes and the ErrorBoundary renders, that's the expected behavior and it works fine, but what if I don't want the Component to crashes but just to render a specific message? In that cause I can't throw an error, I have to send a message to the Front as an error so it can recognize it and render it in the app without crashing, the way I know to do this is by having an error key in the response that is the one that sends an error message if there is one, and then send the appropriate status code in the response. I know I can send custom error message when throwing an error, but then I would have to add the logic to the Action function validating when I want the Component to crash and when I just want to send a custom error message to the Front, which at the end I think it is the same or actually worse error handling.

This are my thoughts of course and I could be wrong, I'm not strong in the Backend haha, so if you know a different way to do it just let me know and maybe we can have a session! We can talk about this in the standup just to clarify our points of view and decide the best way to go.

Again, thanks for the feedback! 👍